src/internal/pfsdb/commits.go

```diff
@@ -0,0 +1,994 @@
+		Size:           commitInfo.Details.SizeBytes,
+		Error:          commitInfo.Error,
+	}
+	namedStmt, err := tx.PrepareNamedContext(ctx, createCommit)
+	if err != nil {
+		return 0, errors.Wrap(err, "prepare create commitInfo")
+	}
+	row := namedStmt.QueryRowxContext(ctx, insert)
+	if row.Err() != nil {
+		if IsDuplicateKeyErr(row.Err()) { // a duplicate key implies that an entry for the repo already exists.
+			return 0, ErrCommitAlreadyExists{CommitID: CommitKey(commitInfo.Commit)}
+		}
+		return 0, errors.Wrap(row.Err(), "exec create commitInfo")
+	}
+	lastInsertId := 0
+	if err := row.Scan(&lastInsertId); err != nil {
+		return 0, errors.Wrap(err, "scanning id from create commitInfo")
+	}
+	if commitInfo.ParentCommit != nil && !opt.SkipParent {
+		if err := CreateCommitParent(ctx, tx, commitInfo.ParentCommit, CommitID(lastInsertId)); err != nil {
+			return 0, errors.Wrap(err, "linking parent")
+		}
+	}
+	if len(commitInfo.ChildCommits) != 0 && !opt.SkipChildren {
+		if err := CreateCommitChildren(ctx, tx, CommitID(lastInsertId), commitInfo.ChildCommits); err != nil {
+			return 0, errors.Wrap(err, "linking children")
+		}
+	}
+	return CommitID(lastInsertId), nil
+}
+
+// CreateCommitParent inserts a single ancestry relationship where the child is known and parent must be derived.
+func CreateCommitParent(ctx context.Context, tx *pachsql.Tx, parentCommit *pfs.Commit, childCommit CommitID) error {
+	ancestryQuery := `
+		INSERT INTO pfs.commit_ancestry
+		(parent, child)
+		VALUES ((SELECT int_id FROM pfs.commits WHERE commit_id=$1), $2)
+		ON CONFLICT DO NOTHING;
+	`
+	_, err := tx.ExecContext(ctx, ancestryQuery, CommitKey(parentCommit), childCommit)
+	if err != nil {
+		if IsParentCommitNotFound(err) {
+			return ErrParentCommitNotFound{ChildRowID: childCommit}
+		}
+		return errors.Wrap(err, "putting commit parent")
+	}
+	return nil
+}
+
+// CreateCommitChild inserts a single ancestry relationship where the parent is known and child must be derived.
+func CreateCommitChild(ctx context.Context, tx *pachsql.Tx, parentCommit CommitID, childCommit *pfs.Commit) error {
+	ancestryQuery := `
+		INSERT INTO pfs.commit_ancestry
+		(parent, child)
+		VALUES ($1, (SELECT int_id FROM pfs.commits WHERE commit_id=$2))
+		ON CONFLICT DO NOTHING;
+	`
+	_, err := tx.ExecContext(ctx, ancestryQuery, parentCommit, CommitKey(childCommit))
+	if err != nil {
+		if IsChildCommitNotFound(err) {
+			return ErrChildCommitNotFound{ParentRowID: parentCommit}
+		}
+		return errors.Wrap(err, "putting commit child")
+	}
+	return nil
+}
+
+// CreateCommitAncestries inserts ancestry relationships where the ids of both parent and children are known.
+func CreateCommitAncestries(ctx context.Context, tx *pachsql.Tx, parentCommit CommitID, childrenCommits []CommitID) error {
+	ancestryQueryTemplate := `
+		INSERT INTO pfs.commit_ancestry
+		(parent, child)
+		VALUES %s
+		ON CONFLICT DO NOTHING;
+	`
+	childValuesTemplate := `($1, $%d)`
+	params := []any{parentCommit}
```

Comment:
Does this work with `tx.ExecContext`?
```suggestion
	params := []CommitID{parentCommit}
```

Reply:
no 😢